doc/internal/man3/OPTIONS.pod

```diff
@@ -0,0 +1,269 @@
+as the specific command from L<openssl(1)> and appends it to the program
+name. This function should only be called once.  Once set, opt_getprog()
+also returns the value.
+
+Once opt_init() has been called, opt_next() can be called in a loop to
+fetch each option in turn. It returns B<-1>, or B<OPT_EOF> when the
+end of arguments has been reached. This is typically done like this:
+
+    prog = opt_init(argc, argv, my_options);
+    while ((o = opt_next()) != OPT_EOF) {
+        switch (o) {
+        case OPT_EOF:
+        case OPT_ERR:
+    opthelp:
+            fprintf(stderr, "%s: Use -help for summary\n", prog);
+            exit(1);
+        case OPT_HELP:
+            opt_help(my_options);
+            exit(0);
+        ...other options...
+        }
+    }
+
+The opt_help() function takes a list of option definitions and prints a
+nicely-formatted output.
+
+Within the option parsing loop, opt_flag() returns the option,
+without any leading hyphens. The opt_arg() function returns
+the option's value, if present.
+
+In an option list, there can be a single option with the empty string.
+This is a "wildcard" or "unknown" option. For example, it allows an
+option to be be taken as digest algorithm, like B<-sha1>. The
+function opt_cipher() takes the specified B<name> and fills in
+the cipher into B<cipherp>.  The function opt_md() does the same
+thing for message digest.
+
+There are a several useful functions for parsing numbers.  Theseare
+opt_int(), opt_long(), opt_ulong(), opt_imax(), and opt_umax().  They all
+take C<0x> to mean hexadecimal and C<0> to mean octal, and will do the
+necessary range-checking. They return B<1> if succcessful and fill in the
+C<result> pointer with the value, or B<0> on error. Note that opt_next()
+will also do range-check on the argument if the appropriate B<valtype>
+field is specified for the option.
+
+The opt_isdir() function returns B<1> if the specified B<name> is
+a directory, or B<0> if not. The opt_format() takes a string value,
+such as used in with the B<-informat> or related flags, and fills
+the value from the constants in F<fmt.h> file.
+
+The opt_string() function checks that B<name> appears in the
+B<NULL>-terminated array of strings. It returns B<1>ne if found,
+or prints a diagnostic and returns B<0> if not.
+
+The opt_pair() function takes a list of B<pairs>, each of which
+has a text name and numeric value. The specified B<name> is
+found on the list, it puts the index in B<*result>, and returns
+B<1>. If not found, it returns B<0>.
+
+After processing all the options, the opt_num_rest() returns what is
+left, and opt_rest() returns a pointer to the first non-option.
+If there were no parameters, it will point to the B<NULL> that is
+at the end of the standard B<argv> array.
+
+=head2 Example
+
+The best examples can be found in sources for the commands in the F<apps>
+directory of the source tree.
+A notable exception is F<apps/cmp.c> which uses this API, but does
```

Comment:
No problem mentioning this.
This CMP app allows putting all options also in a config file in addition to the command line, while the options on the command line can override those given in the config file.